var/spack/repos/builtin/packages/cuda/package.py

```diff
@@ -232,11 +232,13 @@ def install(self, spec, prefix):
             install_shell(*arguments)
 
         # CUDA 10.1+ has different cmdline options for the installer
+        mkdir(join_path(self.stage.path, 'tmp'))  # use stage dir instead of /tmp
         arguments = [
             runfile,            # the install script
             '--silent',         # disable interactive prompts
             '--override',       # override compiler version checks
             '--toolkit',        # install CUDA Toolkit
+            '--tmpdir=%s' % join_path(self.stage.path, 'tmp'),  # use stage dir as /tmp
```

Comment:
- Is this argument supported by all the versions of cuda?  
- Why do you add `/tmp`? 
- The code around line 253 needs to be updated.

Reply:
* good, point, I initially checked just with 9.x, since 10.1 was mentioned in the recipes as having problems with the installer. But I went back all the way and successfully installed:  `cuda@7.5.18 cuda@8.0.61  cuda@9.2.88  cuda@10.1.243  cuda@11.4.4`. The 6.x series fails to install with the argument, so I put an if clause there.
* keeps `spack-stage` neat and tidy :broom: 
* I'm not sure about that - the log is still written to `/tmp`, even if `--tmpdir=` is set, the latter seems to be used just for the extraction. I also don't get the log in the format mentioned in the recipe though, for me cuda seems smart enough to add a random number to it. After installing the above, I have in my tmp:
```
k4-arch-01: /opt/spackages$ ls /tmp/cuda*
/tmp/cuda_install_1822486.log  /tmp/cuda-installer-1822211
/tmp/cuda_install_1822776.log  /tmp/cuda-installer-1822296
/tmp/cuda-installer-1821821    /tmp/cuda-installer-1822876
/tmp/cuda-installer-1821907    /tmp/cuda-installer-1822961
```